kots-exporter/kots-exporter.sh

```diff
@@ -346,8 +343,32 @@ rm_kots_annot_label_resources(){
     echo "Kots removal logs are available - $path/logs/$kotsCleanupLogFile"
 
     echo "Deleting job inject-bottoken-xxxx"
-    kubectl -n $namespace delete "$(kubectl -n $namespace get jobs -o name | grep inject)" \
-    || echo "Manually run the command: kubectl -n $namespace delete $(kubectl -n $namespace get jobs -o name | grep inject)"
+    if [ "$(kubectl -n $namespace get jobs -o name | grep inject)" != "" ]; then
+    kubectl -n $namespace delete "$(kubectl -n $namespace get jobs -o name | grep inject)"
+    fi
+}
+
+postgres_migration(){
+    if "$POSTGRES_INTERNAL"; then
+    echo ""
+    echo "############ PREPARING POSTGRES FOR MIGRATION ############"
+
+    echo "Upgrading to server CircleCI Server 4.0 includes upgrading the Postgres chart"
+    echo "Before upgrading, we need to prepare your postgres instance."
+
+    export POSTGRESQL_PASSWORD="$(kubectl get secret --namespace $namespace postgresql -o jsonpath='{.data.postgresql-password}' | base64 --decode)"
+    export POSTGRESQL_PVC="$(kubectl get pvc --namespace $namespace -l app.kubernetes.io/instance=circleci-server,role=primary -o jsonpath='{.items[0].metadata.name}')"
+    kubectl delete statefulsets.apps postgresql --namespace $namespace --cascade=orphan
+    kubectl delete secret postgresql --namespace $namespace
+
+    echo ""
+    echo "Postgres PVC Name: $POSTGRESQL_PVC"
+    echo ""
+    echo "Creating & tagging Postgresql Secret"
+    kubectl -n $namespace create secret generic postgresql --from-literal=postgres-password="$POSTGRESQL_PASSWORD"
```

Comment:
The postgres secret did not have to be created before, why does it need to create it now?

Reply:
after discussing with @soulchips , It is being created just as a precautionary step and will be overwritten by the helm upgrade command.